### ISSUE TITLE
Socket suddenly stops receiving data

### ISSUE THREAD

**Issue by pRizz** [zeromq/zeromq.js#236]
I am listening on an external ZMQ which works for a couple hours. But then the messages just seem to stop being received. I can verify that the source ZMQ still works as either running another instance of my app or restarting my app does properly receive the queue data.

```
const zmq = require('zeromq')
const sock = zmq.socket('sub')
sock.connect('tcp://redacted')
sock.subscribe('redacted')
sock.on('message', (topic, message) => {
  // do stuff
  // randomly stops after a few hours
})
```

I initially tried listening to an error event and all the monitor events, but the data just stops coming in without any indication as to why. My only clue is that there tends to be a CPU spike on my instance before it stops listening.
```
sock.on('error', (error) => {
    console.log(`${new Date().toISOString()}: An error occurred, `, error)
})

sock.monitor(2000, 0)

// For debugging purposes
sock.on('connect', () => { console.log(`${new Date().toISOString()}: Socket event connect emitted`) })
sock.on('connect_delay', () => { console.log(`${new Date().toISOString()}: Socket event connect_delay emitted`) })
sock.on('connect_retry', () => { console.log(`${new Date().toISOString()}: Socket event connect_retry emitted`) })
sock.on('listen', () => { console.log(`${new Date().toISOString()}: Socket event listen emitted`) })
sock.on('bind_error', () => { console.log(`${new Date().toISOString()}: Socket event bind_error emitted`) })
sock.on('accept', () => { console.log(`${new Date().toISOString()}: Socket event accept emitted`) })
sock.on('accept_error', () => { console.log(`${new Date().toISOString()}: Socket event accept_error emitted`) })
sock.on('close', () => { console.log(`${new Date().toISOString()}: Socket event close emitted`) })
sock.on('close_error', () => { console.log(`${new Date().toISOString()}: Socket event close_error emitted`) })
sock.on('disconnect', () => { console.log(`${new Date().toISOString()}: Socket event disconnect emitted`) })
```

On v4.6.0

**Comment by rolftimmermans** [comment]
Recently we have released 6.0 beta. It features a [new API](https://github.com/zeromq/zeromq.js#examples) that addresses some [fundamental issues](https://github.com/zeromq/zeromq.js/issues/189) with the previous API and also addresses a number of stability bugs. To make upgrading easier it includes a [compatibility layer](https://github.com/zeromq/zeromq.js#compatibility-layer-for-version-45) with versions 4.x/5.x. It would be great if you could give the latest version a spin to see if this solves this particular issue. If you run into any problems with it, feel free to report it here or in a new issue.

**Comment by aminya** [comment]
v6 was released. Please try again with the latest version, and report back if the issue still persists.
https://github.com/zeromq/zeromq.js/releases/tag/v6.0.0
